pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,137 @@
+package testcase
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestSecretsEncrypt() {
+	etcdNodes, errGetEtcd := shared.GetNodesByRoles("etcd")
+	Expect(etcdNodes).NotTo(BeEmpty())
+	Expect(errGetEtcd).NotTo(HaveOccurred(), "error getting etcd nodes")
+
+	cpNodes, errGetCP := shared.GetNodesByRoles("control-plane")
+	Expect(cpNodes).NotTo(BeEmpty())
+	Expect(errGetCP).NotTo(HaveOccurred(), "error getting control plane nodes")
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	ips := getNodeIps(etcdNodes, cpNodes)
+
+	errSecret := shared.CreateSecret("secret1", "default")
+	Expect(errSecret).NotTo(HaveOccurred(), "error creating secret")
+
+	secretsEncryptOps("prepare", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("rotate", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("reencrypt", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("rotate-keys", product, cpNodes[0].ExternalIP, ips)
+}
+
+func secretsEncryptOps(action, product, cpIp string, ips []string) {
+	shared.LogLevel("INFO", fmt.Sprintf("TEST: Secrets-Encryption: %s", action))
+	_, errStatusB4 := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatusB4).NotTo(HaveOccurred(), "error getting secret-encryption status before action")
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		time.Sleep(20 * time.Second)
+	}
+
+	for _, node := range ips {
+		var nodearr []string
+		nodearr = append(nodearr, node)
+		nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
+		Expect(errRestart).NotTo(HaveOccurred(), "error restart service for node: "+nodeIp)
+		// Order of reboot matters. Etcd first then control plane nodes.
+		// Little lag needed between node restarts to avoid issues.
+		shared.LogLevel("INFO", "Sleep 10 seconds - wait before restarting next node in cluster")
+		time.Sleep(10 * time.Second)
+	}
+	switch product {
+	case "k3s":
+		shared.LogLevel("INFO", "Sleep 30 seconds - wait for services to come up")
+		time.Sleep(30 * time.Second)
+	case "rke2":
+		shared.LogLevel("INFO", "Sleep 60 seconds - wait for services to come up")
+		time.Sleep(60 * time.Second)
+	}
+
+	stdStatusOut, errStatus := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatus).NotTo(HaveOccurred(), "error getting secret-encryption status")
+	verifyStatusOutput(action, stdStatusOut)
+
+	errLog := logEncryptionFileContents(ips, product)
+	Expect(errLog).NotTo(HaveOccurred())
+}
+
+func verifyStdOut(action, stdout string) {
```

Comment:
same comment- 


can we return error please ? since has some possible areas that could go wrong, would be good to send the error above